app/services/sso_service.rb

```diff
@@ -17,7 +17,9 @@ class SSOService
   def initialize(response)
     raise 'SAML Response is not a OneLogin::RubySaml::Response' unless response.is_a?(OneLogin::RubySaml::Response)
     @saml_response = response
-
+    # rubocop:disable Style/RescueModifier
+    Raven.tags_context(saml_issuer: @saml_response&.issuers) rescue nil
```

Comment:
Why rescue nil?

Reply:
Also, I don't think `issuers` is the same as the `authn_context`.

I thought `authn_context` comes from: `REXML::XPath.first(saml_response.decrypted_document, '//saml:AuthnContextClassRef')&.text`

Reply:
authn_context is nil for id.me right?  I was wary that all sever errors would be considered id.me errors, which could be misleading.  

I used `rescue nil` because if params[:SAMLResponse] is ever nil we get `OneLogin::RubySaml::ValidationError: Issuer of the Response not found or multiple` and in that case we want it to continue with typical error checks.

Reply:
What are all the different permutations of issuers? I've just never seen that before.

And no, i think authn_context as we have it defined is nil, but the REXML will point to something like an id.me URL when it is ID.me

Reply:
We have a raise error up above if the saml response is ever nil or invalid, so I don't know if rescue nil is going to serve the intended purpose.

Reply:
when I log in with id.me, the saml issuer is ["api.idmelabs.com"].  i'm not sure about the others, since I'm not able to log in to those in dev. 
I was only considering this a fallback to the antun_context on line 142.  It can go.

Reply:
I take back what I said about deleting this.  I think this is the closest thing we have to an identifier. 
Grabbing the [the saml response from sentry](http://sentry.vetsgov-internal/vets-gov/platform-api-production/issues/978/events/1821905/) for "Login Fail! Subject did not consent to attribute release" it's the most useful thing I could find in the XML.



````

<?xml version="1.0" encoding="UTF-8"?>
<samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="_c1539f949dab408b8cfe55d9957a0b41" Version="2.0" IssueInstant="2018-12-03T16:20:26Z" Destination="https://api.va.gov/auth/saml/callback" InResponseTo="_a4bc8f6f-f20e-45a5-91b1-ab45b1931f0b">
  <saml:Issuer>idp.id.me</saml:Issuer>
  <ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
    <ds:SignedInfo>
      <ds:CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#" />
      <ds:SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256" />
      <ds:Reference URI="#_c1539f949dab408b8cfe55d9957a0b41">
        <ds:Transforms>
          <ds:Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature" />
          <ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#" />
        </ds:Transforms>
        <ds:DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256" />
        <ds:DigestValue>[DELETED DIGEST]</ds:DigestValue>
      </ds:Reference>
    </ds:SignedInfo>
    <ds:SignatureValue>[DELETED SIG]</ds:SignatureValue>
    <ds:KeyInfo>
      <ds:X509Data>
        <ds:X509Certificate>[DELETED CERT]</ds:X509Certificate>
      </ds:X509Data>
    </ds:KeyInfo>
  </ds:Signature>
  <samlp:Status>
    <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Responder">
      <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:AuthnFailed" />
    </samlp:StatusCode>
    <samlp:StatusMessage>Subject did not consent to attribute release</samlp:StatusMessage>
  </samlp:Status>
</samlp:Response>
````

Reply:
... but sentry server has a bug where most of the saml response are truncated and look like this:
<img width="287" alt="mvi_find_profile_error__soap_http_call_failed" src="https://user-images.githubusercontent.com/19188/49403180-03d06700-f71a-11e8-9ecf-6f39dc824d45.png">
Combine that with the fact that I can only log in with id.me in dev and this is still a bit of a guess if it's useful.